File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    visited = [0] * n
    for i in range(n):
        if not visited[i]:
            queue = deque([i])
            visited[i] = 1
            while queue:
                next_node = queue.popleft()
                for j in range(n):
                    if not visited[j] and computers[next_node][j] == 1:
                        visited[j] = 1
                        queue.append(j)
            answer += 1
            
            
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;
    static Deque<Integer> queue;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        queue = new ArrayDeque<>();
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                bfs(i, computers);
                answer++;
            }
        }
        
        return answer;
    }
    
    static void bfs(int start, int[][] computers) {
        
        queue.add(start);
        visited[start] = true;
        while(!queue.isEmpty()) {
            int curr = queue.poll();
            
            for (int next = 0 ; next < computers.length; next++) {        if (computers[curr][next] == 1 && !visited[next]) {
                    visited[next] = true;
                    queue.add(next);
                }
                
            }
        }
        
        
    }
}